extensions/amp-script/0.1/amp-script.js

```diff
@@ -649,15 +671,24 @@ export class SanitizerImpl {
             dev().error(TAG, 'Invalid AMP.setState() argument: %s', value);
           });
           if (state) {
-            // Only evaluate updates in case of recent user interaction.
-            const skipEval = !this.userActivationTracker_.isActive();
-            if (skipEval) {
+            const fullEval = this.allowFullEval_();
+            const constrainedEval = this.allowConstrainedEval_();
+
+            if (!fullEval && !constrainedEval) {
               user().warn(
                 TAG,
                 'AMP.setState only updated page state and did not reevaluate bindings due to lack of recent user interaction.'
               );
             }
-            bind.setState(state, skipEval, /* skipAmpState */ false);
+            bind.setState(
+              state,
+              /* skipEval */ !fullEval,
+              /* skipAmpState */ false
+            );
+
+            if (constrainedEval) {
+              bind.constrainedEvaluation([this.element_]);
```

Comment:
This needs to be chained after `setState` completes, though I'd consider making it a single parameterized method call rather than actually doing `.then()` to keep the API cleaner.

Reply:
Looking around at the various `apply` functions in `bind-impl`, I don't see a super clean way to work into this.

What do you think about a new parameter in `bind.setState`? 
```js
function setState(state, { skipEval, skipAmpState, constrainToElements }) 
```

Reply:
SGTM

Reply:
Done in https://github.com/ampproject/amphtml/pull/26067